pkg/validation/app.go

```diff
@@ -536,3 +535,26 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+func isClusterSingleton(entry v1alpha1.AppCatalogEntry, cr, app v1alpha1.App) bool {
+	if entry.Spec.Restrictions.ClusterSingleton {
+		if key.IsInOrgNamespace(cr) &&
+			key.ClusterID(cr) == key.ClusterID(app) ||
+			cr.Spec.KubeConfig.Context.Name == app.Spec.KubeConfig.Context.Name {
+			return true
+		}
+		if !key.IsInOrgNamespace(cr) {
+			return true
+		}
+	}
+	return false
+}
+
+func isNamespaceSingleton(entry v1alpha1.AppCatalogEntry, cr, app v1alpha1.App) bool {
+	if entry.Spec.Restrictions.NamespaceSingleton {
```

Comment:
how about you return entry.Spec.Restrictions.NamespaceSingleton && app.Spec.Namespace == cr.Spec.Namespace instead of this whole if?